File: Schedule/Schedule.Core/Models/Speciality.cs
namespace Schedule.Core.Models;

public class Speciality
{
    public int SpecialityId { get; set; }

    public string Code { get; set; } = null!;

    public string Name { get; set; } = null!;

    public int MaxTermId { get; set; }

    public bool IsDeleted { get; set; }

    public virtual ICollection<Discipline> Disciplines { get; set; } = new List<Discipline>();

    public virtual ICollection<Group> Groups { get; set; } = new List<Group>();

    public virtual Term MaxTerm { get; set; } = null!;
}

File: schedule.ios.applicaion/schedule.ios.applicaion/Models/Speciality.swift
//
//  Speciality.swift
//  schedule.ios.application
//
//  Created by Иван Светлов on 11.06.2023.
//

import Foundation

struct Speciality: Identifiable, Codable, Equatable, Hashable {
    var id: Int
    var code: String
    var name: String
    var maxTermId: Int
    var isDeleted: Bool
}
